### PR TITLE
Improved MarkdownViewer

### PR DESCRIPTION
**Pull request by KamilDev**
#### Features
- Uses the latest [github-markdown.css](https://github.com/sindresorhus/github-markdown-css/blob/main/github-markdown.css) file that contains styling support for both dark and light mode.
- Table of Contents has an improved design, and:
	- The width can be resized.
	- TOC headings are automatically highlighted to help track your position in the file.
- Improved fullscreen layout based on Github.com behavior.
- External links clicked will open in default browser instead of doing nothing.
- Uses [markdownItAnchor](https://github.com/valeriangalliat/markdown-it-anchor) to allow heading anchor links in the file to work.
- Uses [highlight.js](https://github.com/highlightjs/highlight.js) to provide syntax highlighting to codeblocks.
#### Changes
- Made changes to allow the `md2html.html` file to use relative file imports for better maintainability.
- MarkdownViewer can now easily be customized by users by modifying files in `<Quicklook data folder>/QuickLook.Plugin.MarkdownViewer/`
- Caching and `localStorage` is now supported thanks to these changes.
- Prevent default behavior of spacebar scrolling the page, while we use spacebar to dismiss the preview.
- Sets `WebView` `DefaultBackgroundColor` to prevent white flash in dark mode. After the page has loaded, sets `DefaultBackgroundColor` back to white to have the expected default behavior on HTML pages that don't specify any background color.
#### Clean up
- Removed the need for `jQuery`.
- Removed old polyfill code.

#### Screenshots
New:
Screenshot 2024-12-04 214042

Old:
Screenshot 2024-12-04 223338

### CODE REVIEW

File: QuickLook.Plugin/QuickLook.Plugin.MarkdownViewer/Plugin.cs [hunk 1]
```diff
@@ -54,29 +132,180 @@ public void View(string path, ContextObject context)
...
-            _panel.NavigateToHtml(GenerateMarkdownHtml(path));
+            var htmlPath = GenerateMarkdownHtml(path);
+            _panel.NavigateToFile(htmlPath);
             _panel.Dispatcher.Invoke(() => { context.IsBusy = false; }, DispatcherPriority.Loaded);
+
+            _panel._webView.NavigationStarting += NavigationStarting_CancelNavigation;
         }
 
-        public void Cleanup()
+        private void NavigationStarting_CancelNavigation(object sender, CoreWebView2NavigationStartingEventArgs e)
```
Comment (emako) [inline QuickLook.Plugin/QuickLook.Plugin.MarkdownViewer/Plugin.cs:142]: Can you move to class WebpagePanel?

File: QuickLook.Plugin/QuickLook.Plugin.MarkdownViewer/Plugin.cs [hunk 2]
```diff
@@ -54,29 +132,180 @@ public void View(string path, ContextObject context)
             context.ViewerContent = _panel;
             context.Title = Path.GetFileName(path);
 
-            _panel.NavigateToHtml(GenerateMarkdownHtml(path));
+            var htmlPath = GenerateMarkdownHtml(path);
+            _panel.NavigateToFile(htmlPath);
             _panel.Dispatcher.Invoke(() => { context.IsBusy = false; }, DispatcherPriority.Loaded);
+
+            _panel._webView.NavigationStarting += NavigationStarting_CancelNavigation;
```
Comment (emako) [inline QuickLook.Plugin/QuickLook.Plugin.MarkdownViewer/Plugin.cs:139]: This line throw an exception:
System.FieldAccessException: 方法“QuickLook.Plugin.MarkdownViewer.Plugin.View(System.String, QuickLook.Common.Plugin.ContextObject)”尝试访问字段“QuickLook.Plugin.HtmlViewer.WebpagePanel._webView”失败。
   在 QuickLook.Plugin.MarkdownViewer.Plugin.View(String path, ContextObject context) 位置 D:\Github\QuickLook\QuickLook.Plugin\QuickLook.Plugin.MarkdownViewer\Plugin.cs:行号 139
   在 QuickLook.ViewerWindow.<>c__DisplayClass6_0.<BeginShow>b__1() 位置 D:\Github\QuickLook\QuickLook\ViewerWindow.Actions.cs:行号 222
System.FieldAccessException: 方法“QuickLook.Plugin.MarkdownViewer.Plugin.View(System.String, QuickLook.Common.Plugin.ContextObject)”尝试访问字段“QuickLook.Plugin.HtmlViewer.WebpagePanel._webView”失败。
   在 QuickLook.Plugin.MarkdownViewer.Plugin.View(String path, ContextObject context) 位置 D:\Github\QuickLook\QuickLook.Plugin\QuickLook.Plugin.MarkdownViewer\Plugin.cs:行号 139
   在 QuickLook.ViewerWindow.<>c__DisplayClass6_0.<BeginShow>b__1() 位置

File: QuickLook.Plugin/QuickLook.Plugin.HtmlViewer/WebpagePanel.cs [hunk 3]
```diff
@@ -29,8 +30,8 @@ namespace QuickLook.Plugin.HtmlViewer
 {
     public class WebpagePanel : UserControl
     {
-        private Uri _currentUri;
-        private WebView2 _webView;
+        public Uri _currentUri;
```
Comment (emako) [inline QuickLook.Plugin/QuickLook.Plugin.HtmlViewer/WebpagePanel.cs:33]: If the new `NavigationStarting_CancelNavigation` moved to WebpagePanel, public should be restore to private.

**Review by emako — CHANGES_REQUESTED** [review]

File: QuickLook.Plugin/QuickLook.Plugin.MarkdownViewer/Plugin.cs (hunk 2 above)
Comment (KamilDev) [reply QuickLook.Plugin/QuickLook.Plugin.MarkdownViewer/Plugin.cs]: I'm unable to reproduce this exception. Please see if it's resolved in the newest commit.

**Comment by emako** [comment]
@KamilDev  Could you provide some markdown file to test the code about `// Open in default browser`?

**Comment by KamilDev** [comment]
Here's a test file with some links:
[example.md](https://github.com/user-attachments/files/18037373/example.md)

The Microsoft Store and Spotify links should work, for a real example. The rest are used to see that the app name associated with the URI is correct. If no app is installed that handles the URI, or it's invalid, Windows will show the native message that no app is installed to handle the URI.

The behavior should be the same when viewing .html, since the code is now in `WebpagePanel`.

**Pull request merged**